lib/api/apiUtils/bucket/bucketDeletion.js

```diff
@@ -151,12 +152,23 @@ export function deleteBucket(bucketMD, bucketName, canonicalID, log, cb) {
                     return _deleteMPUbucket(bucketName, log, next);
                 });
         },
+        function destroyBucketKey(next) {
```

Comment:
@LaurenSpiegel are you sure the bucket may be silently recreated on object PUT? the only usage of BucketInfo::removeDeletedFlag I saw is made from bucketCreation.js.

BTW you're right, the bucket encryption key must last as long as the flagged bucket exists in the metadata backend